zetaclient/zetabridge.go

```diff
@@ -192,9 +194,10 @@ func (b *ZetaCoreBridge) UpdateConfigFromCore(cfg *config.Config, init bool) err
 	if err != nil {
 		b.logger.Info().Msg("Unable to fetch keygen from zetacore")
 	}
+
 	cfg.UpdateCoreParams(keyGen, newChains, newEVMParams, newBTCParams, init, b.logger)
+	cfg.SetKeygen(keyGen)
```

Comment:
I think keyGen was already set in line '198' inside 'UpdateCoreParams'.

Reply:
This one is using a lock which fixes the issue